### PR TITLE
:bug: Fix `ClusterRole`, `ClusterRoleBinding` names and `app.kubernetes.io/instance` label

### PR DESCRIPTION
**Pull request by LorianColtof**
### What does this PR do?

1. It adds the name of the namespace to `ClusterRole` and `ClusterRoleBinding` names.
2. It appends the namespace to `app.kubernetes.io/instance` label.

### Motivation

This is to prevent `ClusterRole` and `ClusterRoleBinding` resources with the same names when there are multiple releases with the same name but in different namespaces.

### More

- [X] Yes, I updated the tests accordingly
- [X] Yes, I ran `make test` and all the tests passed

### CODE REVIEW

**Comment by mloiseleur** [comment]
Hello @LorianColtof 

Thanks for your PR.  It's interesting and it's also adding complexity for an edge case. 

What do you think about 
1. Adding `app.kubernetes.io/instance: {{ .Release.Name }}-{{ .Release.Namespace }}` for all objects in the existing `labels` helper ? 
2. Set name of ClusterRole and ClusterRoleBinding to `{{- template "traefik.name" . -}}-{{ .Release.Namespace }}` instead of creating a specific helper ?

**Comment by LorianColtof** [comment]
Hi @mloiseleur,

I agree with 1, wasn't entirely sure what to do with the `app.kubernetes.io/instance` label. For 2, this theoretically gives a problem when there are multiple chart installations in the same namespace. What do you think about `{{- template "traefik.fullname" . -}}-{{ .Release.Namespace }}`?

**Comment by mloiseleur** [comment]
@LorianColtof 
> What do you think about {{- template "traefik.fullname" . -}}-{{ .Release.Namespace }}?

Yes, it makes sense, since it's current name is based on fullname.

File: traefik/templates/_helpers.tpl [hunk 1]
```diff
@@ -47,6 +47,19 @@ The name of the service account to use
...
+Adds the namespace to name to prevent duplicate resource names when there
+are multiple namespaced releases with the same release name.
+*/}}
+{{- define "traefik.clusterRoleName" -}}
+{{- if .Values.rbac.namespaced -}}
+{{- template "traefik.fullname" . -}}-{{ .Release.Namespace }}
+{{- else -}}
+{{- template "traefik.fullname" . -}}
+{{- end -}}
+{{- end -}}
```
Comment (mloiseleur) [inline traefik/templates/_helpers.tpl:61]:
```suggestion
{{- define "traefik.clusterRoleName" -}}
{{- template "traefik.fullname" . -}}-{{ .Release.Namespace }}
{{- end -}}
```

It's quite unexpected to have a different name when you enable a feature. 
And this issue can also happens without `rbac.namespaced`.

**Review by mloiseleur — CHANGES_REQUESTED** [review]

**Comment by mloiseleur** [comment]
Thanks @LorianColtof 
I left a last suggestion. After that, we should probably be able to merge it.

**Comment by LorianColtof** [comment]
I applied your suggestion @mloiseleur.

**Review by mloiseleur — APPROVED** [review]

**Review by charlie-haley — APPROVED** [review]

**Review by darkweaver87 — APPROVED** [review]
LGTM

File: traefik/templates/_helpers.tpl [hunk 2]
```diff
@@ -37,7 +37,7 @@ If release name contains chart name it will be used as a full name.
 app.kubernetes.io/name: {{ template "traefik.name" . }}
 helm.sh/chart: {{ template "traefik.chart" . }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
-app.kubernetes.io/instance: {{ .Release.Name }}
+app.kubernetes.io/instance: {{ .Release.Name }}-{{ .Release.Namespace }}
```
Comment (Project0) [inline traefik/templates/_helpers.tpl:40]: Could this potentially break something? I also do not see why it needs a suffix as the (helm) release name should be already unique per namespace.

Comment (LorianColtof) [reply traefik/templates/_helpers.tpl:40]: According to the Kubernetes documentation this label should contain "A unique name identifying the instance of an application". So for the global objects `ClusterRole` and `ClusterRoleBinding` this would not be unique without the suffix if there were multiple releases with the same name but in different namespaces. I do agree though that the suffix is not necessary for the namespaced objects, which is why I initially used a separate label helper for `ClusterRole` and `ClusterRoleBinding`, but maybe it is nice to make this label consistent throughout all objects like @mloiseleur suggested.

Comment (mloiseleur) [reply traefik/templates/_helpers.tpl:40]: @LorianColtof Discussing this change on `app.kubernetes.io/instance` label, we see potentially a breaking change here and we are not sure of the use case. Can you detail us more why you also need this change ? or do you think we are good to merge this PR without this label change ?

Comment (LorianColtof) [reply traefik/templates/_helpers.tpl:40]: I don't need this label change in particular. I added it just to keep this label unique for the `ClusterRole`/`ClusterRoleBinding` cases.

Comment (mloiseleur) [reply traefik/templates/_helpers.tpl:40]: @LorianColtof Thanks. It's more clear. We bumped to a Major version, since it can breakj something for people using this label.
